onelogin_aws_cli/__init__.py

```diff
@@ -75,6 +82,24 @@ def get_saml_assertion(self):
 
         self.saml = saml_resp
 
+    def get_ip_address(self) -> Optional[str]:
```

Comment:
I would propose that this should be a `@property` method in `onelogin_aws_cli.Section`.

eg,

    @property
    def ip_address(self):
      """..."""
      return ipify.get_ip() \
        if self.do_ip_address_auto_discover \
        else self.get('ip_address', None)

and then called with `section.get('ip_address')`

I've put a PR together to put forward a consistent method of accessing configs related to this in

Reply:
It doesn't really make sense to me to have external network calls in something called `configuration.Section` to be honest. I would expect a class like that to only parse config data.

Reply:
Arguably it is config data. 😄 The functionality makes an external call to retrieve an option which can be defined in the config file.

Reply:
Semantically, sure it can be considered config data. But practically, you want to be able to parse a config file without depending on external things over a network. This is a best practice that for example improves testability.

Reply:
I think it's ok here, but we can discuss further once we get @drewsonne 's config refactor done =).